### PR TITLE
Add support for Continuous Delivery

### LINKED ISSUE #3: Deploy with Docker
Create a Dockerfile in the root directory with configuration for building a image.

Might consider pushing the Docker image to a team member's Docker Hub account for easy accessibility?

### PR DESCRIPTION
**Pull request by kappsegla**
Adds multistage Dockerfile for building dockerimage. Adds spring-boot-starter-actuator to pom.xml to enable health checks. delivery.yml github action for automatic deployments.

Changes to application.properties
Enable virtual threads
Set forward-headers-strategy=framework to enable X- headers behind gateway. Add environment variables with default values for. DATASOURCE_URL, DATASOURCE_USERNAME and DATASOURCE_PASSWORD

Closes #3

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -1,2 +1,6 @@
-spring.graphql.graphiql.enabled=true
-
+spring.datasource.url=${DATASOURCE_URL:jdbc:mysql://localhost:3306/mmotodo}
+spring.datasource.username=${DATASOURCE_USERNAME:root}
+spring.datasource.password=${DATASOURCE_PASSWORD:root}
```
Comment (saeraa) [inline src/main/resources/application.properties:3]: Did not know you could do this to supply default environment variables. Very handy!

**Review by saeraa — APPROVED** [review]
Looking good!

**Review by ved0 — APPROVED** [review]
Great work. Keep it up! 👍

**Review by solidtop — APPROVED** [review]

File: src/main/resources/application.properties (hunk 1 above)
Comment (kappsegla) [reply src/main/resources/application.properties:3]: Really neat to get it to work both in development/production and not needing different profiles. Or just as an option for some developers in the team to change the values without changing the properties file and making conflicts when merging commits.

**Pull request merged**